examples/asr/asr_hybrid_transducer_ctc/helpers/convert_nemo_asr_hybrid_to_ctc.py

```diff
@@ -0,0 +1,170 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+A script to convert a Nemo ASR Hybrid model file (.nemo) to a Nemo ASR CTC or RNNT model file (.nemo)
+
+This allows you to train a RNNT-CTC Hybrid model, but then convert to a pure CTC or pure RNNT model for use
+in Riva. Works just fine with nemo2riva, HOWEVER, Riva doesn't support AggTokenizer, but nemo2riva
+does, so be careful that you do not convert a model with AggTokenizer and then use that in Riva
+as it will not work.
+
+Usage: python convert_nemo_asr_hybrid_to_ctc.py -i /path/to/hybrid.nemo -o /path/to/saved_ctc_model.nemo -m [ctc|rnnt]
+
+"""
+
+
+import argparse
+import os
+
+import torch
+from omegaconf import OmegaConf
+
+from nemo.collections.asr.models import (
+    ASRModel,
+    EncDecCTCModel,
+    EncDecCTCModelBPE,
+    EncDecRNNTBPEModel,
+    EncDecRNNTModel,
+)
+from nemo.utils import logging
+
+
+def extract_model_ctc(args, hybrid_model):
+    BPE = False
+    ctc_class = EncDecCTCModel
+    if 'tokenizer' in hybrid_model.cfg.keys():
+        BPE = True
+        ctc_class = EncDecCTCModelBPE
+
+    hybrid_model_cfg = OmegaConf.to_container(hybrid_model.cfg)
+
+    new_cfg = hybrid_model_cfg.copy()
+    new_cfg['ctc_reduction'] = hybrid_model_cfg['aux_ctc']['ctc_reduction']
+    new_cfg['decoder'] = hybrid_model_cfg['aux_ctc']['decoder']
+    del new_cfg['compute_eval_loss']
+    del new_cfg['model_defaults']
```

Comment:
In your opinion, which style is better?
To copy explicitly pieces of hybrid config to new config 

Or

To copy entire hybrid config and delete components not belongin to ctc. 

Since you've worked with both styles, your opinion is preferred on which one is a practical approach.

Reply:
I honestly think the copy+delete approach is probably best for easier maintenance going forward. That way, if we add new features to models, such as inter ctc or variational noise, they will be preserved and flow through. If we go with a select-based approach, then someone would regularly have to keep an eye on this script to ensure it's still selecting everything necessary as future architectural changes happen.

Reply:
Makes sense, lets go with the delete approach then.